### DISCUSSION TITLE
How to declare files to be uploaded without UploadFile?

### DISCUSSION THREAD

**Discussion by peterbandi** [fastapi/fastapi#11309] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import fastapi
import streaming_form_data
import streaming_form_data.targets

app = fastapi.FastAPI()

@app.post(path="/upload")
async def upload(request: fastapi.Request):
    file_target = streaming_form_data.targets.FileTarget(filename="/upload/input")
    form_parser = streaming_form_data.StreamingFormDataParser(headers=request.headers)
    form_parser.register(name="file", target=file_target)

    async for chunk in request.stream():
        form_parser.data_received(chunk)
```

### Description

I would like to declare files to be uploaded, but without `UploadFile`. My problem is that the files to be uploaded are very large, about 3GiB to 4GiB. This holds up the server for way too long. Additionally, after the file has been downloaded and written to disk, I have to copy the file to a different location anyway.

The problem is solved by streaming the `Request` object manually to the destination. Unfortunately, in this case I don't know how to declare files to be uploaded so it would also show up in the OpenAPI documentation generated by Swagger.

### Operating System

Linux

### Operating System Details

Ubuntu on WLS2 on Windows 11: Ubuntu 20.04.6 LTS (GNU/Linux 5.15.146.1-microsoft-standard-WSL2 x86_64)

### FastAPI Version

0.109.2

### Pydantic Version

2.6.1

### Python Version

Python 3.9.18

**Comment by YuriiMotov** [comment, marked as answer]
You can add the schema using `openapi_extra` parameter:

```python
from fastapi import FastAPI

app = FastAPI()

@app.get(
    "/",
    openapi_extra={
        "requestBody": {
            "content": {
                "multipart/form-data": {
                    "schema": {
                        "properties": {
                            "file": {
                                "type": "string",
                                "format": "binary",
                                "title": "File",
                            }
                        },
                        "type": "object",
                        "required": ["file"],
                    }
                }
            },
            "required": True,
        },
    },
)
async def root():
    return {"message": "Hello World"}

```

This way file parameter will be added to openapi schema.
But remember that you are in charge of request validation
